### PR TITLE
Fixes Lazy Templates fucking over cables, pipes , and shuttles at random

### PR DESCRIPTION
**Pull request by ZephyrTFA**
## About The Pull Request

See title.
## Why It's Good For The Game

If a lazy template is loaded AFTER roundstart it doesn't get included into the global init procs, because that work is offloaded into a subsystem
LINDA runtimes are bad, null.archive() runtime be gone.
## Changelog
:cl:
fix: The Syndicate have fired their previous construction company after poor results in recent outposts.
/:cl:

### CODE REVIEW

**Comment by LemonInTheDark** [comment]
Not an acceptable solution. You should not need to manhandle active turf lists, this problem SHOULD be handled in ChangeTurf
Similar for initializing cables/pipes/shuttles. These cases should already be supported, this is a papering over of existing issues.

**Comment by ninjanomnom** [comment]
At least as far as the pipes and cables, this seems to be a pattern from 2017 or earlier, but even then it was a mistake. I would need to look more into it to be sure but I think it could be resolved by queuing an update within init with the subsystem then having that subsystem handle the updates next time it ticks.

File: code/modules/shuttle/shuttle.dm [hunk 1]
```diff
@@ -239,6 +239,9 @@
 		for(var/turf/T in return_turfs())
 			T.turf_flags |= NO_RUINS
 
+	if(SSshuttle.initialized)
+		SSshuttle.setup_shuttles(list(src))
```
Comment (ZephyrTFA) [inline code/modules/shuttle/shuttle.dm:243]:
```suggestion
		ASYNC
			SSshuttle.setup_shuttles(list(src))
```

File: code/datums/lazy_template.dm [hunk 2]
```diff
@@ -64,12 +64,21 @@
 	if(!reservation)
 		CRASH("Failed to reserve a block for lazy template: '[key]'")
 
+	// lists kept for overall loading
 	var/list/loaded_atom_movables = list()
 	var/list/loaded_turfs = list()
 	var/list/loaded_areas = list()
+
+	// list which matches a type to a list of things for that type
+	var/list/type_to_list = list(
```
Comment (LemonInTheDark) [inline code/datums/lazy_template.dm:73]: do we really need this level of extensibility here? has overhead in a pretty hot loop

File: code/datums/lazy_template.dm [hunk 3]
```diff
@@ -81,12 +90,25 @@
...
+						continue
+					type_to_list[load_type] += thing
+					break
+
 				loaded_atom_movables |= thing
 
-	SSatoms.InitializeAtoms(loaded_atom_movables + loaded_turfs + loaded_areas)
+	SSatoms.InitializeAtoms(loaded_areas)
+	SSatoms.InitializeAtoms(loaded_atom_movables)
+	SSatoms.InitializeAtoms(loaded_turfs)
```
Comment (LemonInTheDark) [inline code/datums/lazy_template.dm:108]: why separate?

**Review by LemonInTheDark — CHANGES_REQUESTED** [review]

File: code/datums/lazy_template.dm (hunk 3 above)
Comment (ZephyrTFA) [reply code/datums/lazy_template.dm:108]: I had issues locally where it kept initializing stuff in the wrong order, it needs to be A -> AM -> T for undertile stuff to work correctly

Comment (ZephyrTFA) [reply code/datums/lazy_template.dm:108]: not sure if it's just my local Byond being stupid tho

File: code/game/turfs/change_turf.dm [hunk 4]
```diff
@@ -211,9 +211,12 @@ GLOBAL_LIST_INIT(blacklisted_automated_baseturfs, typecacheof(list(
...
 				stashed_group.display_turf(new_turf)
 	else
+		for(var/turf/open/adjacent_turf as anything in atmos_adjacent_turfs)
+			adjacent_turf.atmos_adjacent_turfs -= src
+		atmos_adjacent_turfs = null
 		if(excited || excited_group)
 			SSair.remove_from_active(src) //Clean up wall excitement, and refresh excited groups
-		if(ispath(path,/turf/closed) || ispath(path,/turf/cordon))
+		if(ispath(path, /turf/closed) || ispath(path, /turf/cordon))
 			flags |= CHANGETURF_RECALC_ADJACENT
```
[the remark this answers is not in the document]
Comment (ZephyrTFA) [reply code/game/turfs/change_turf.dm:220]: When the map loader processes, it calls ChangeTurf with the CHANGETURF_DEFER_CHANGE flag, which means AfterChange never gets called which is what handles that flag
AfterChange is manually called once everything is done, however, due to the CHECK_TICKS that happen inside loading this means that SSair is both able to run, and process on turfs adjacent to the build area

File: code/datums/lazy_template.dm (hunk 3 above)
Comment (LemonInTheDark) [reply code/datums/lazy_template.dm:102]: What it's doing before should be fine? InitializeAtoms isn't magic, so what you describe doesn't really make sense

File: code/game/turfs/change_turf.dm (hunk 4 above)
[the remark this answers is not in the document]
Comment (LemonInTheDark) [reply code/game/turfs/change_turf.dm:220]: It makes me real uncomfortable I'ma be honest. if I'm honest I'm sorta confused about how anything is active at all by this point, is it part of the changeturf logic?

I do not like just doing recalc_adjacent but more sus directly in changeturf. Smells really wrong

File: code/datums/lazy_template.dm (hunk 3 above)
Comment (LemonInTheDark) [reply code/datums/lazy_template.dm]: for the record I could see something like late init needing to happen before another init, but that'd be a bug tbh

File: code/game/turfs/change_turf.dm (hunk 4 above)
[the remark this answers is not in the document]
Comment (LemonInTheDark) [reply code/game/turfs/change_turf.dm:220]: Lemme see about how to do this better, will get back to you (hopefully)

**Review by Ghommie — APPROVED** [review]

**Pull request merged**